### DISCUSSION TITLE
Dynamic bleopt prompt_rps1

### DISCUSSION THREAD

**Discussion by Strykar** [akinomyoga/ble.sh#503] [Q&A] [answered]
Hi, how can I can get the RPS to show me (full path cwd) of non-git directory on the right side and show git details only when inside a git repo directory?
Currently, I can only get it to work with one, not both:
```
bleopt complete_menu_complete=1
bleopt complete_auto_complete=1
bleopt complete_menu_filter=1
bleopt prompt_rps1='\q{contrib/git-info}'
bleopt prompt_rps1_transient=1
ble-sabbrev clean_lpt_journal='| grep -vE "http: TLS handshake error from|block_watcher\.go:[0-9]+] Polling blocks from=" "$@"'

function my/set-up-completion {
  bleopt complete_auto_delay=300
}
ble-import core-complete -C 'my/set-up-completion'

function blerc/define-sabbrev-branch {
  function blerc/sabbrev-git-branch {
    ble/util/assign-array COMPREPLY "git branch | sed 's/^\*\{0,1\}[[:space:]]*//'" 2>/dev/null
  }
  ble-sabbrev -m '\branch'=blerc/sabbrev-git-branch
}
blehook/eval-after-load complete blerc/define-sabbrev-branch
ble-import contrib/prompt-git

function my/vim-load-hook {
  ((_ble_bash>=40300)) && builtin bind 'set keyseq-timeout 1'
  ble-import vim-airline
  bleopt vim_airline_theme=dark
  source "$_ble_base/lib/vim-surround.sh"
}
blehook/eval-after-load keymap_vi my/vim-load-hook
```

**Comment by akinomyoga** [comment, marked as answer]
You can define your own prompt sequence as described in [§4.13.2](https://github.com/akinomyoga/ble.sh/wiki/Manual-%C2%A74-Editing#4132-user-defined-prompt-sequences). For the specific setup of switching between the current working directory and the git information based on the current directory, [Recipe 4](https://github.com/akinomyoga/ble.sh/wiki/Recipes#r4-show-current-working-directory-and-git-branch-in-right-prompt) actually provides an example. Recipe 4 implements the git information by itself, but this example is old. We now have a module `contrib/prompt-git`, so one can set it up more easily using `contrib/prompt-git`:

```bash
# blerc

ble-import contrib/prompt-git

function ble/prompt/backslash:Strykar/git-info {
  local "${_ble_contrib_prompt_git_vars[@]/%/=}"
  if ble/contrib/prompt-git/initialize; then
    # Inside a git repository
    ble/prompt/process-prompt-string '\q{contrib/git-info}'
  else
    # In a non-git directory
    ble/prompt/process-prompt-string '\w'
  fi
}
bleopt prompt_rps1='\q{Strykar/git-info}'
```

**Comment by Strykar** [comment]
Thank you so much!
I had tried https://github.com/akinomyoga/ble.sh/wiki/Recipes#r4-show-current-working-directory-and-git-branch-in-right-prompt at first and then I moved to `contrib/prompt-git` once I realized it was newer, like in my example above.

IMO, the example should be updated with your's above.

Now I can free `PS1` of `cwd` reporting duty!

**Comment by Strykar** [comment]
@akinomyoga One last request, could the example be updated to use symbols for even more git information, like:
- https://github.com/twolfson/sexy-bash-prompt
- https://github.com/twolfson/sexy-bash-prompt/blob/master/screenshot.png

Possible for RPS to be shown in Powerline style?

**Reply by akinomyoga** [reply]
As described at the bottom of ["Features" section in README](https://github.com/akinomyoga/ble.sh?tab=readme-ov-file#features), ble.sh intends to offer an underlying framework for the prompt settings. It doesn't intend to provide specific prompt themes. I'd leave the detailed themes to other frameworks or the users' customizations. The recipes and examples just demonstrate how the prompt themes can be designed for the users who want to configure it by themselves and the theme developers. Once you understand how it works, you should be able to add symbols by yourself.

Or you can actually modify the `sexy-bash-prompt` setting for ble.sh. Instead of assigning a value to `PS1`, you can assign the related part to `bleopt prompt_rps1`. I think the following part is related:

https://github.com/twolfson/sexy-bash-prompt/blob/c6acbd323f2e0a2fe266fd0cde5865218ef0a6c3/.bash_prompt#L327-L331
